addons/mail/static/src/new/discuss/sidebar.xml

```diff
@@ -21,6 +21,7 @@
             </t>
         </div>
         <hr class="w-100 border-top"/>
+        <input t-if="Object.values(messaging.state.threads).filter(thread => thread.is_pinned and thread.model === 'mail.channel').length > 19" class="form-control mx-4 mb-2 rounded-3 w-auto" placeholder="Quick search..." t-model="state.quickSearchVal"/>
```

Comment:
condition is a bit too complex for my taste to be on a template

Reply:
introduced getter `hasQuickSearch`